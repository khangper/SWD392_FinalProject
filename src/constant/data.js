export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
