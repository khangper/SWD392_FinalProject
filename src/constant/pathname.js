export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
