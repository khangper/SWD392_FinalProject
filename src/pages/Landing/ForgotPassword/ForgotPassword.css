.reset-container {
    margin-top: -75px;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
  }
  
  .reset-container::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: url('https://gambolthemes.net/html-items/cursus-new-demo/images/sign.svg'); 
    background-size: cover;
    background-position: center;
    opacity: 0.06; 
    z-index: -1;
  }
  
  .reset-logo-container {
    text-align: center;
    margin-bottom: 45px;

  }
  
  .reset-logo {
    height: 30px;
  }
  
  .reset-form-container {
    background: white;
    padding: 45px;
    border-radius: 8px;
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
    text-align: center;
    width: 560px;
    position: relative;
    z-index: 1;
    padding-bottom: 5px;
  }
  
  h2 {
    margin: 0;
    margin-bottom: 45px;
  }
  .reset-input-wrapper {
    position: relative;
    width: 100%;
    margin-bottom: 35px;
  }
  .reset-input-icon {
    position: absolute;
    left: 10px;
    top: 50%;
    transform: translateY(-50%);
    width: 20px;
    height: 20px; 
    color: #666;
  }
  
  .reset-input-field {
    width: 100%;
    padding: 10px;
    padding: 10px 10px 10px 40px; 
    border: 1px solid #ddd;
    border-radius: 4px;
    font-size: 14px;
  }
  
  .reset-btn {
    width: 100%;
    padding: 10px;
    margin-bottom: 10px;
    border: none;
    border-radius: 4px;
    font-size: 14px;
    cursor: pointer;
  }
  
  .reset-password {
    background: #ff0000;
    color: white;
    margin-bottom: 35px;
  }
  .reset-password:hover {
    background: rgb(21, 21, 21);
  }
  
  .back-to-login {
    display: flex;
    justify-content: center;
    gap: 5px;
    margin-top: 10px;
  }
  
  .back-to-login p{
    color: #666;
    font-size: 14px;
  }
  .sign-in-link {
    color: #007bff;
  text-decoration: none;
  font-size: 14px;
  }
  
  .sign-in-link:hover {
    text-decoration: none;
    color: #666;
  }
  
  .form-footer {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-top: 40px;
    margin-bottom: 30px;
  }
  
  .footer-logo {
    height: 30px;
    margin-bottom: 10px;
    margin-right: 10px;
  }
  
  .form-footer span {
    color: #000000;
    font-size: 14px;
  }