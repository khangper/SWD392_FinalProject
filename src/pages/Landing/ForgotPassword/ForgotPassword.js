import React from "react";
import { Link } from "react-router-dom";
import "./ForgotPassword.css";
import email from "../../../assets/email.png";
import hakaFull_logo from "../../../assets/Haka_Online_Learning_System_logo.png"

const ForgotPassword = () => {
  return (
    <div className="reset-container">
      <div className="reset-logo-container">
        <Link to="/home">
          <img
            src={hakaFull_logo}
            alt="Cursus Logo"
            className="reset-logo"
          />{" "}
        </Link>
      </div>
      <div className="reset-form-container">
        <h2>Request a Password Reset</h2>
        <div className="reset-input-wrapper">
          <img src={email} alt="Email" className="reset-input-icon" />
          <input
            type="email"
            placeholder="Email Address"
            className="reset-input-field"
          />
        </div>
        <button className="reset-btn reset-password">Reset Password</button>
        <div className="back-to-login">
          <p>Go Back</p>
          <Link to="/sign-in" className="sign-in-link">
            Sign In
          </Link>
        </div>
      </div>
      <div className="form-footer">
        <img
          alt=""
          src={hakaFull_logo}
          className="footer-logo"
        />
        <span>
          Â© 2024 <strong>Haka</strong>. All Rights Reserved.
        </span>
      </div>
    </div>
  );
};

export default ForgotPassword;
