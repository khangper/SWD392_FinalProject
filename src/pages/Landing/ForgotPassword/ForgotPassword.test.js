import React from "react";
import { renderWithProviders } from "../../../constant/test-util";
import ForgotPassword from "./ForgotPassword";
import { screen } from "@testing-library/react";

describe("ForgotPassword", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the ReportHistory component", async () => {
        const { container } = renderWithProviders(<ForgotPassword />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})

//   it("displays the main title correctly", () => {
//     renderWithProviders(<ReportHistory />);
//     const titleElement = screen.getByText("Report history");
//     expect(titleElement).toBeInTheDocument();
//   });

//   it("displays the thank you message correctly", () => {
//     renderWithProviders(<ReportHistory />);
//     const thankYouMessage = screen.getByText("Thanks for reporting");
//     expect(thankYouMessage).toBeInTheDocument();
//   });

//   it("contains a link to Community Guidelines", () => {
//     renderWithProviders(<ReportHistory />);
//     const linkElement = screen.getByText("Community Guidelines");
//     expect(linkElement).toBeInTheDocument();
//     expect(linkElement.closest("a")).toHaveAttribute("href", "#");
//   });

//   it("contains a link to learn more about reporting content", () => {
//     renderWithProviders(<ReportHistory />);
//     const learnMoreLink = screen.getByText(
//       "Learn more about reporting content on Cursus."
//     );
//     expect(learnMoreLink).toBeInTheDocument();
//     expect(learnMoreLink.closest("a")).toHaveAttribute("href", "#");
//   });

//   it("displays the message for no submitted reports", () => {
//     renderWithProviders(<ReportHistory />);
//     const noReportsMessage = screen.getByText(
//       "You haven't submitted any reports."
//     );
//     expect(noReportsMessage).toBeInTheDocument();
//   });
// });
