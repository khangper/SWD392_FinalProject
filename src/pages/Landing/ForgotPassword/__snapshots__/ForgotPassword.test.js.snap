// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword renders the ForgotPassword component 1`] = `
<div>
  <div
    class="reset-container"
  >
    <div
      class="reset-logo-container"
    >
      <a
        href="/home"
      >
        <img
          alt="Cursus Logo"
          class="reset-logo"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
        />
         
      </a>
    </div>
    <div
      class="reset-form-container"
    >
      <h2>
        Request a Password Reset
      </h2>
      <div
        class="reset-input-wrapper"
      >
        <img
          alt="Email"
          class="reset-input-icon"
          src="email.png"
        />
        <input
          class="reset-input-field"
          placeholder="Email Address"
          type="email"
        />
      </div>
      <button
        class="reset-btn reset-password"
      >
        Reset Password
      </button>
      <div
        class="back-to-login"
      >
        <p>
          Go Back
        </p>
        <a
          class="sign-in-link"
          href="/sign-in"
        >
          Sign In
        </a>
      </div>
    </div>
    <div
      class="form-footer"
    >
      <img
        alt=""
        class="footer-logo"
        src="https://gambolthemes.net/html-items/cursus-new-demo/images/sign_logo.png"
      />
      <span>
        © 2024 
        <strong>
          Cursus
        </strong>
        . All Rights Reserved.
      </span>
    </div>
  </div>
</div>
`;

exports[`ForgotPassword renders the ReportHistory component 1`] = `
<div>
  <div
    class="reset-container"
  >
    <div
      class="reset-logo-container"
    >
      <a
        href="/home"
      >
        <img
          alt="Cursus Logo"
          class="reset-logo"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
        />
         
      </a>
    </div>
    <div
      class="reset-form-container"
    >
      <h2>
        Request a Password Reset
      </h2>
      <div
        class="reset-input-wrapper"
      >
        <img
          alt="Email"
          class="reset-input-icon"
          src="email.png"
        />
        <input
          class="reset-input-field"
          placeholder="Email Address"
          type="email"
        />
      </div>
      <button
        class="reset-btn reset-password"
      >
        Reset Password
      </button>
      <div
        class="back-to-login"
      >
        <p>
          Go Back
        </p>
        <a
          class="sign-in-link"
          href="/sign-in"
        >
          Sign In
        </a>
      </div>
    </div>
    <div
      class="form-footer"
    >
      <img
        alt=""
        class="footer-logo"
        src="https://gambolthemes.net/html-items/cursus-new-demo/images/sign_logo.png"
      />
      <span>
        © 2024 
        <strong>
          Cursus
        </strong>
        . All Rights Reserved.
      </span>
    </div>
  </div>
</div>
`;
